### PR TITLE
PFS-717: update barebones demo to demonstrate root custodial transactions

### PR DESCRIPTION
**Pull request by developer-darpan**
## https://futureverse.atlassian.net/browse/PFS-717 https://futureverse.atlassian.net/browse/PFS-647

- use config from env.vars so we can deploy this on vercel easily.
- add support for Root  EVM precompile based tx signing.
- use UUID for CS clientIds
- add apple, discord, roblox login buttons

### CODE REVIEW

File: src/pages/callback/transactions.ts [hunk 1]
```diff
@@ -130,13 +142,27 @@ export async function signTransaction(
...
+
   const signTransactionPayload = {
     account: fromAccount,
     transaction: serializedUnsignedTransaction,
+    idpUrl: identityProviderUri,
     callbackUrl,
   }
 
-  const id = 'client:2'
+  const id = `client:${uuidV4()}`
```
[the remark this answers is not in the document]
Comment (developer-darpan) [reply src/pages/callback/transactions.ts:165]: its just a string type [reference](https://github.com/futureversecom/futurepass/blob/732ef70a02ee2cda0634cc026520d5e97d6bcce5/libs/rpc-kit/src/lib/RPCClient.ts#L83)

have checked it working with staging CS without any issues.

File: src/config.ts [hunk 2]
```diff
@@ -1,21 +1,69 @@
...
 
 // Used for mixpanel tracking
-export const mixpanelProjectToken = 'dfb084e3ff87f5aed5d21cc88ec38a7e'
+export const mixpanelProjectToken = getOrThrowFromEnv('MIXPANEL_PROJECT_TOKEN')
+
+export const rootJsonRpcProviderUrl = getOrThrowFromEnv('ROOT_JSON_RPC_URL')
+export const rootReceiverAddress = getOrThrowFromEnv('ROOT_RECEIVER_ADDRESS')
+export const rootChainId = getOrThrowFromEnv('ROOT_CHAIN_ID')
+
+export const ERC20_ABI = [
```
Comment (kabece) [inline src/config.ts:33]: Isn't this exported from Ethers?

**Review by kabece — APPROVED** [review]
LGTM

File: src/config.ts (hunk 2 above)
Comment (developer-darpan) [reply src/config.ts:33]: have defined the ABI here,
so it would be easier for developers to extend this example to interact with their custom contracts.

**Pull request merged**